compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1894,6 +1894,24 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             if let ty::Array(..) | ty::Slice(..) = ty.kind() {
                 err.help("the semantics of slice patterns changed recently; see issue #62254");
             }
+        } else if expected_ty.is_adt() {
+            if expected_ty.ty_adt_def()
+                == self
+                    .tcx
+                    .get_diagnostic_item(sym::vec_type)
+                    .and_then(|vec_ty| self.tcx.type_of(vec_ty).ty_adt_def())
+            {
+                if let (Some(span), true) = (ti.span, ti.origin_expr) {
+                    if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(span) {
+                        err.span_suggestion(
+                            span,
+                            "consider slicing here",
+                            format!("{}[..]", snippet),
+                            Applicability::MachineApplicable,
+                        );
+                    }
+                }
+            }
```

Comment:
Can you use `AutoDeref::new().find()` to see if any type derefs to slice?

Reply:
Fixed, thanks for the suggestion!

Reply:
Nice! I wasn't sure if that would work. You can replace `find().is_some()` with `any()`.